File: app/src/main/java/com/material/androidpro/dashboard/DashboardFragment.java
package com.material.androidpro.dashboard;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.material.androidpro.R;
import com.material.androidpro.databinding.FragmentDashboardBinding;

public class DashboardFragment extends Fragment {

    public DashboardFragment() {
        // Required empty public constructor
    }

    public static DashboardFragment newInstance() {
        DashboardFragment fragment = new DashboardFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        FragmentDashboardBinding binding = DataBindingUtil.inflate(inflater,
                R.layout.fragment_dashboard, container, false);

        binding.recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));

        DashboardViewModel viewModel = new DashboardViewModel(getActivity());
        binding.setVm(viewModel);

        return binding.getRoot();
    }
}

File: app/src/main/java/com/material/androidpro/dashboard/DashboardFragment.kt
package com.material.androidpro.dashboard

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.LinearLayoutManager
import com.material.androidpro.R
import com.material.androidpro.base.BaseFragment
import com.material.androidpro.databinding.FragmentDashboardBinding

class DashboardFragment : BaseFragment() {

    private lateinit var mBinding: FragmentDashboardBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        mBinding = DataBindingUtil.inflate(
            inflater,
            R.layout.fragment_dashboard, container, false
        )

        with(mBinding) {
            recyclerView.layoutManager = LinearLayoutManager(context)
        }

        val viewModel = DashboardViewModel(requireActivity())
        mBinding.vm = viewModel

        return mBinding.root
    }

}